src/link/filelink.cpp

```diff
@@ -24,6 +25,11 @@ FileLink::FileLink(QObject* parent)
 
 void FileLink::_writeData(const QByteArray& data)
 {
+    // Check if we have already openned the file
```

Comment:
Why can't we just use the flag isOpen to determine if the file needs to be opened? This patch appears obfuscating to me (spaghetti). Right now, it seems like a better solution is to avoid creating the filelink until we are ready to write data.

Reply:
Ok, I'll rewrite this patch. But I disagree to move this issue in pingviewer. The Links need to abstract everything related to connection or creation, special cases does not need to be handle in the graphic interface.

Reply:
Ok. Maybe there is the case though where you want an empty file created, maybe not. I agree with you.

Reply:
Logic redone, I think that is much more clear now, thanks for the feedback.